### ISSUE TITLE
Feature: Support more Models out of the box

### ISSUE THREAD

**Issue by milafrerichs** [tzachar/cmp-ai#26]
Thanks for you package and your feedback on my PR.

Regarding my PR #25 I put some more thought in and wanted to create an issue to keep the discussion more visible and not in an existing PR.

## Where am I coming from
I want to use a different model than the default you set. To do so I need to adjust the `prompt` function and of course change the model. 

So instead of the default I need to look up what the template for the Model is and adjust the prompt.
Template: 
```
{{- if .Suffix }}<|fim_prefix|>{{ .Prompt }}<|fim_suffix|>{{ .Suffix }}<|fim_middle|>
{{- else }}{{ .Prompt }}
{{- end }}
```

So I would write the options as follows:
```lua
provider_options = {
	model = 'codegemma:2b-code',
	prompt = function(lines_before, lines_after)
		return '<|fim_prefix|>' .. lines_before .. '<|fim_suffix|>' .. lines_after .. '<|fim_middle|>'
	end,
}
```

And if I want to use a different model I would need to do the same.

## Proposal
Support more models by not having a default prompt for a specific model but a more general purpose prompt and use the existing templates via a `suffix` param. 

Instead of having a default prompt for a specific model we just return the lines_before
```diff
- prompt = self.params.prompt and self.params.prompt(lines_before, lines_after) or '<PRE> ' .. lines_before .. ' <SUF>' .. lines_after .. ' <MID>',
+ prompt = self.params.prompt and self.params.promt(lines_before, lines_after) or lines_before
+ suffix = lines_after
```

This results in the exact same behaviour the plugin has currently. Since the default models template looks like this:
```
{{- if .Suffix }}<PRE> {{ .Prompt }} <SUF>{{ .Suffix }} <MID>
{{- else }}{{ .Prompt }}
{{- end }}
```

But now every other model can be used without touching `prompt` or anything else. Someone just needs to adjust the modelname. 

## Future Improvements
The plugin could introduce a new option to disable the suffix.   
For example the `raw` parameter could be introduced which would be a nice addition as well since it would override the template. 

From the ollama docs:
`raw`: if true no formatting will be applied to the prompt. You may choose to use the raw parameter if you are specifying a full templated prompt in your request to the API

I'm happy to adjust my PR or create a new one.

**Comment by tzachar** [comment]
closed by #25

**Comment by milafrerichs** [comment]
hey @tzachar 
this not really closed by #25 
this is an additional/different proposal. 

It is about changing the default prompt implementation:

https://github.com/tzachar/cmp-ai/blob/61d86ffc5863833aa30c6098f95a1e56670a6c64/lua/cmp_ai/backends/ollama.lua#L23

to: 
```lua
prompt = self.params.prompt and self.params.promt(lines_before, lines_after) or lines_before
suffix = lines_after
```

**Comment by tzachar** [comment]
Ok, can you create a different PR?
